### DISCUSSION TITLE
Test strategy

### DISCUSSION THREAD

**Discussion by ycaoT** [twilio-labs/paste#1671] [Office hours] [answered]
Does paste team have test strategy can share, like the paste-nextjs-template. It helps people using Paste components, and gives them a way to follow.

**Comment by gloriliale** [comment, marked as answer]
Hi Yang! After discussing this with the team, it does seem like this is a question better suited for the larger Twilio frontend community as opposed to Office Hours. We'd suggest asking in one of these:  #help-javascript #eng-guild-frontend #eng-frontend-forum or #help-testing

We don't have a suggested testing strategy for consumers at the moment.

**Reply by SiTaggart** [reply]
I guess the only other thing I could add here is just listing out the stack.

- Jest and React Testing Library for component tests
- aXe Jest for components accessibility tests
- Cypress for E2E / integration testing

It's a pretty common testing stack at Twilio.

We also use Chromatic for visual regression testing of our components and UI review, which runs off of Storybook.

**Reply by ycaoT** [reply]
@SiTaggart  I want to see how does Paste setup these, maybe something can share with me to use. Just like the paste-nextjs-template

**Reply by SiTaggart** [reply]
Setting up a consuming apps testing environment isn't very sharable unfortunately.

It's so unique to the consuming apps current architecture and stack. Some use commonJS, some use esModules, some JS and some Typescript. You might need to tweak each tools config and setup to match your own stack. We can really only recommend the tools.

Our code is obviously open source for you to look at, but we do some very specific things because we're running a monorepo. 

You might also get a bunch of information searching the internal GitHub enterprise for examples on how other teams are setting up their testing environments when using Paste.

Sorry, we can't be much more help than that for now.
